lib/runners/processor/fxcop.rb

```diff
@@ -20,68 +16,42 @@ def analyzer_version
       ENV.fetch('FXCOP_VERSION')
     end
 
-    def setup
-      # restore project (install dependencies)
-      _, _, status = capture3('dotnet', 'restore')
-      unless status.success?
-        msg = <<~EOS
-          Failed to restore .NET Core Project. You have to put project file (.csproj) under the analysis root. Please check the project structure and 'root_dir' parameter in #{config.path_name}.
-          See: https://help.sider.review/getting-started/custom-configuration#root_dir-option
-          Note: We only support a project managed with MSBuild. No support for other build systems. (e.g. Nuke, Cake)
-        EOS
-        return Results::Failure.new(guid: guid, message: msg, analyzer: analyzer)
-      end
-
-      # install FxCop Analyzers from nuget Repository
-      capture3!('dotnet', 'add', 'package', 'Microsoft.CodeAnalysis.FxCopAnalyzers', '--version', analyzer_version)
-
-      yield
-    end
-
     def analyze(changes)
-      # run 'dotnet build' with static analysis module
       output_file = Tempfile.create(["fxcop-", ""]).path
-      capture3!('dotnet', 'build', '--no-incremental', "-property:errorlog=#{output_file}")
-      output_json = read_output_json(output_file)
+      capture3!('Sider.RoslynAnalyzersRunner',
+        '--outputfile', output_file,
+        *changes
+          .changed_paths
+          .select{|p| p.extname.eql?(".cs")}
+          .map{|p| relative_path(working_dir / p, from: current_dir)}
+          .map(&:to_s))
```

Comment:
[suggestion] What about the following refactoring idea?

```suggestion
          .filter_map{|f| relative_path(working_dir / f, from: current_dir).to_s if f.extname == ".cs"})
```

- [`filter_map`](https://ruby-doc.org/core-2.7.1/Enumerable.html#method-i-filter_map) is useful in this case.
- Use `f` instead of the confusing [`p`](https://ruby-doc.org/core-2.7.1/Kernel.html#method-i-p) method.
- Use `==` instead of [`eql?`](https://ruby-doc.org/core-2.7.1/Object.html#method-i-eql-3F), because the former is used usually (both is different).